### ISSUE TITLE
Disabling shallow clone is no longer supported in version 7.0.0

### ISSUE THREAD

**Issue by steliosfran** [bitrise-steplib/steps-git-clone#199]
### Troubleshooting

- [X] I've searched [discuss.bitrise.io](https://discuss.bitrise.io/) for possible solutions.
- Which version of the step is effected? **7.0.0**
- Is the issue reproducible with the latest version? **YES**
- Does the issue happen sporadically, or every time? **EVERY TIME**

### Useful information

- [FAQ](https://devcenter.bitrise.io/faq/faq-index/)
- [devcenter.bitrise.io](https://devcenter.bitrise.io/)
- [Shallow clone by default release notes](https://github.com/bitrise-steplib/steps-git-clone/releases/tag/7.0.0)

### Issue description

Disabling shallow clone is no longer supported. Previously, a `clone_depth` of `0` was disabling shallow clone.

### Steps to reproduce

1. Under `Clone options`, set a `Clone depth` of `0`
2. Run the step

**Expected outcome**: `git fetch` uses no depth
**Actual outcome**: `git fetch` uses a depth of `1`

**Comment by ofalvai** [comment]
@steliosfran by "depth of 0" do you mean literally adding the `--depth=0` flag to the git commands? Because 0 is not a valid value, the smallest depth is 1.

```
$ git clone --depth=0 ...
fatal: depth 0 is not a positive number
```

**Comment by DriesH** [comment]
Would be nice if we could disable the depth somehow, I quickly checked the code and it seems that if you set the depth to 0 in the input values of the step it will [choose the ideal depth for you](https://github.com/bitrise-steplib/steps-git-clone/blob/master/gitclone/gitclone.go#L55) (which is [1 or 50](https://github.com/bitrise-steplib/steps-git-clone/blob/f6f56378327c4d237889db8ebbc2a85174b8f530/gitclone/checkout.go#L334)).

We have a custom step later in our workflow that uses the commit count as the value for the patch version of our build version: `x.x.GIT_CLONE_COMMIT_COUNT`. (Not the most ideal solution, I know... 😅)

**Comment by steliosfran** [comment]
> @steliosfran by "depth of 0" do you mean literally adding the `--depth=0` flag to the git commands? Because 0 is not a valid value, the smallest possible depth is 1.
> 
> ```
> $ git clone --depth=0 ...
> fatal: depth 0 is not a positive number
> ```

@ofalvai You're right, it's not supported by git.

In the previous versions of this step though, a value of `0` was disabling the depth.

How can we disable it in version 7.0.0?

**Comment by jacobhzen** [comment]
There needs to be an option to disable shallow clone entirely. This breaks our mainline builds because they use commit count for build number. We have to revert to an earlier version to work around this.

**Comment by dongheeleeme** [comment]
Yes, it would be good if it supports disabling the shallow clone option. the new version should support backward compatibility for many builds which is using commit count.

**Comment by ofalvai** [comment]
👋 everyone - we released version [7.0.1](https://github.com/bitrise-steplib/steps-git-clone/releases/tag/7.0.1) with the option to disable depth limiting completely. If you really want to fetch the entire repo history, just set `-1` as the `clone_depth` input.
